server/api/auth_handler.py

```diff
@@ -0,0 +1,168 @@
+                if auth_token:
+                    g.user = user
+                    responseObject = {
+                        "status": "success",
+                        "message": "Successfully logged in",
+                        "auth_token": auth_token.decode()
+                    }
+                    resp = make_response(jsonify(responseObject))
+                    resp.set_cookie("Authentication token", auth_token.decode(), httponly=True)  # Should the auth_token be encoded or decoded?
+                    return resp, 200
+                else:
+                    responseObject = {
+                        "status": "fail",
+                        "message": "User does not exist."
+                    }
+                    return make_response(jsonify(responseObject)), 404
+            elif user:
+                responseObject = {
+                    "status": "fail",
+                    "message": "Password is incorrect."
+                }
+                return make_response(jsonify(responseObject)), 401
+            else:
+                responseObject = {
+                    "status": "fail",
+                    "message": "User does not exist."
+                }
+                return make_response(jsonify(responseObject)), 404
+        except Exception as e:
+            responseObject = {
+                "status": "fail",
+                "message": "Try again"
+            }
+            return make_response(jsonify(responseObject)), 500
+
+
+class UserAPI(MethodView):
+    def get(self):
+        auth_header = request.headers.get("Authorization")
+        if auth_header:
+            auth_token = auth_header.split(" ")[1]
+        else:
+            auth_token = ""
+        if auth_token:
+            resp = User.decode_auth_token(auth_token)
+            if not isinstance(resp, str):
+                user = User.query.filter_by(id=resp).first()
+                responseObject = {
+                    "status": "success",
+                    "data": {
+                        "user_id": user.id,
+                        "email": user.email,
+                        "registered_time": user.registered_time
+                    }
+                }
+                return make_response(jsonify(responseObject)), 200
+            responseObject = {
+                "status": "fail",
+                "message": resp
+            }
+            return make_response(jsonify(responseObject)), 401
+        else:
+            responseObject = {
+                "status": "fail",
+                "message": "Provide a valid auth token."
+            }
+            return make_response(jsonify(responseObject)), 401
+
+
+# Basic middleware for protected routes pulled from Flask's documentation.
+def login_required(f):
+    @wraps(f)
+    def decorated_function(*args, **kwargs):
+        if g.user is None:
+            return redirect(url_for('login', next=request.url))
+        return f(*args, **kwargs)
```

Comment:
Here you  check if the user is in the "g" object. Instead, you'll want to check if the tokens in the cookie is valid. Note that the value of "g" is wiped between requests, it's only used for storing data temporarily within the context of a single request. You can use it to add the user object after you validate the token, though. You can take the user id, find the user and add it to g. Then it can be used in the route